snovault/tests/test_schemas.py

```
@@ -31,3 +47,349 @@ def test_schemas_etag(testapp):
+    ak_strings = ['/profiles/access_key.json', '/profiles/access_key',
+                  'access_key.json', 'access_key']
+    other_strings = ['profiles/access_key.jsonbgood', '']
+
+    for s in ak_strings:
+        assert name == _get_item_name_from_schema_id(s)
+    for os in other_strings:
+        assert name != _get_item_name_from_schema_id(os)
+
+
+def test_is_submittable_schema_given_item_name(schema_for_testing):
+    with composite_mocker_for_schema_utils(sub_item_names=['tester']):
+        ans = _is_submittable_schema(schema_for_testing.get('$id'), schema_for_testing.get('properties'))
+        assert ans is True
+
+
+def test_is_not_submittable_schema_given_item_name(schema_for_testing):
+    with composite_mocker_for_schema_utils(sub_item_names=['not_tester']):
+        ans = _is_submittable_schema(schema_for_testing.get('$id'), schema_for_testing.get('properties'))
+        assert ans is False
+
+
+def test_is_submittable_schema_given_key_prop(schema_for_testing):
+    with composite_mocker_for_schema_utils(sub_prop='submitted_id'):
+        ans = _is_submittable_schema(schema_for_testing.get('$id'), schema_for_testing.get('properties'))
+        assert ans is True
+
+
+def test_is_not_submittable_schema_wo_key_prop(schema_for_testing):
+    with composite_mocker_for_schema_utils(sub_prop='other_id'):
+        ans = _is_submittable_schema(schema_for_testing.get('$id'), schema_for_testing.get('properties'))
+        assert ans is False
+
+
+def test_is_not_submittable_schema_wo_info(schema_for_testing):
+    with composite_mocker_for_schema_utils():
+        ans = _is_submittable_schema(schema_for_testing.get('$id'), schema_for_testing.get('properties'))
+        assert ans is False
+
+
+def test_submittable_no_app_info(testapp):
+    """ without providing item name or key prop we expect
+        a 200 response but no json
+    """
+    test_uri = '/can-submit/access_key.json'
+    res = testapp.get(test_uri, status=200)
+    assert not res.json
```

Perhaps this should raise an HTTP error rather than indicate success? On a similar note, should probably raise an HTTP error if someone requests an item type that does not have a schema, which I think would currently result in the same response here that doesn't really provide information. Consider adding a test for this situation to state the behavior more explicitly.

---

I will look into this in the context of raising exceptions in certain situations.  Thanks for the suggestion.